### PR TITLE
ocamlPackages.alcotest: 0.8.5 → 1.0.1

### PR DESCRIPTION
**Pull request by vbgl**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vbgl** [comment]
@GrahamcOfBorg build stog jackline

File: pkgs/development/ocaml-modules/base64/default.nix [hunk 1]
```diff
@@ -13,9 +13,16 @@ buildDunePackage {
 
   minimumOCamlVersion = "4.03";
 
-  buildInputs = [ alcotest bos ];
+  buildInputs = [ bos ];
+
+  # Fix test-suite for alcotest ≥ 1.0
+  patches = [(fetchpatch {
+    url = "https://github.com/mirage/ocaml-base64/commit/8d334d02aa52875158fae3e2fb8fe0a5596598d0.patch";
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/base64/default.nix:20]: Why not update `base64` to [3.4.0](https://github.com/mirage/ocaml-base64/releases/tag/v3.4.0) which contains this commit?

File: pkgs/development/ocaml-modules/base64/default.nix [hunk 2]
```diff
@@ -13,9 +13,16 @@ buildDunePackage {
...
+    sha256 = "0lvqdp98qavpzis1wgwh3ijajq79hq47898gsrk37fpyjbrdzf5q";
+  })];
 
   doCheck = true;
+  checkInputs = lib.optional doCheck alcotest;
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/base64/default.nix:25]: Not sure, but doesn't `mkDerivation` already [remove `checkInputs`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix#L136) for you if `doCheck` is disabled?

File: pkgs/development/ocaml-modules/digestif/default.nix [hunk 3]
```diff
@@ -12,6 +12,9 @@ buildDunePackage rec {
     sha256 = "09g4zngqiw97cljv8ds4m063wcxz3y7c7vzaprsbpjzi0ja5jdcy";
   };
 
+  # Fix tests with alcotest ≥ 1
+  patches = [ ./test_conv_ml.patch ];
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/digestif/default.nix:16]: You could use `fetchpatch` here: <https://github.com/mirage/digestif/commit/8f69623ecccbd6e18cad97e42124615d9a771179>

Comment (vbgl) [reply pkgs/development/ocaml-modules/digestif/default.nix]: Done.

File: pkgs/development/ocaml-modules/base64/default.nix (hunk 2 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/base64/default.nix]: You are right. Thanks for the hint.

File: pkgs/development/ocaml-modules/base64/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/base64/default.nix:20]: Maybe next PR…

**Pull request merged**